### PR TITLE
Add fallback to no VCS when VCS not supported, and support Mercurial

### PR DESCRIPTION
**Pull request by dchenk**
1. If fossa-cli discovers a VCS that is not supported, it should warn the user and fallback to the no-VCS method.
2. Add support for Mercurial as a VCS.

### CODE REVIEW

File: vcs/vcs.go [hunk 1]
```diff
@@ -73,9 +73,9 @@ func GetRepository(dirname string) (string, error) {
 		return "", err
 	}
 	switch vcs {
-	case Git, Subversion, None:
+	case Git, Subversion, Mercurial, None:
 		return dir, nil
-	case Mercurial, Bazaar:
+	case Bazaar:
```
[the remark this answers is not in the document]
Comment (dchenk) [reply vcs/vcs.go]: Actually `GetRepository` isn't used anywhere. Perhaps it was needed for something in the past, but now it's redundant. Should I just remove it?

File: vcs/mercurial.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	dir     string
+	project string
+	head    Revision
+}
+
+// NewMercurialRepository uses the Mercurial repository's metadata at dir to identify the codebase.
+func NewMercurialRepository(dir string) (*MercurialRepository, error) {
+	cmd, _, err := exec.Which("--version", os.Getenv("HG_BINARY"), "hg")
+	if err != nil {
+		return nil, errors.Wrap(err, "could not find Mercurial binary")
```
[the remark this answers is not in the document]
Comment (dchenk) [reply vcs/mercurial.go]: Fair enough. For now it's probably best to keep behavior across all our supported VCS implementations consistent. There's also the problem that, if we fail to run `git` or `svn` or something like this and fall back to constructing the needed information some other way, then inattentive users who don't notice or understand a warning message may assume that the CLI is being buggy when they see their project appearing different all of a sudden in the fossa.com dashboard.

File: vcs/vcs.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (zlav) [reply vcs/vcs.go]: Yea thats interesting, feel free to remove it if you find that it's unnecessary.

File: vcs/mercurial_test.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewMercurialRepository(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skip integration test")
```
[the remark this answers is not in the document]
Comment (dchenk) [reply vcs/mercurial_test.go]: I thought the container in which we run unit tests doesn't have the VCS tools installed, no?

[the remark this answers is not in the document]
Comment (zlav) [reply vcs/mercurial_test.go]: Yea thats exactly why I had second thoughts once I realized you were running `hg`. I would make sure that `hg` is installed in that image as well. It looks like it is if the tests are passing.

File: vcs/vcs.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (dchenk) [reply vcs/vcs.go]: Cool. I'll convert the tests in `vcs_test` from testing `GetRepository` to testing `Nearest`, which is what they should've been testing.

**Pull request merged**